File: Server/src/main/content/global/skill/production/cooking/recipe/pizza/impl/PineapplePizza.java
package content.global.skill.production.cooking.recipe.pizza.impl;

import content.global.skill.production.cooking.recipe.pizza.PizzaRecipe;
import core.game.interaction.NodeUsageEvent;
import core.game.node.item.Item;

/**
 * The Pineapple pizza.
 */
public class PineapplePizza extends PizzaRecipe {
    private static final Item PINEAPPLE_PIZZA = new Item(2301);
    private static final Item PINEAPPLE_RING = new Item(2118);
    private static final Item PINEAPPLE_CHUNKS = new Item(2116);

    @Override
    public double getExperience() {
        return 52;
    }

    @Override
    public int getLevel() {
        return 65;
    }

    @Override
    public Item getProduct() {
        return PINEAPPLE_PIZZA;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{PINEAPPLE_CHUNKS, PINEAPPLE_RING};
    }

    @Override
    public String getMixMessage(NodeUsageEvent event) {
        return "You add the " + event.getBaseItem().getName().toLowerCase() + " to the pizza.";
    }

}

File: Server/src/main/content/global/skill/production/cooking/recipe/pizza/impl/PineapplePizza.kt
package content.global.skill.production.cooking.recipe.pizza.impl

import content.global.skill.production.cooking.recipe.pizza.PizzaRecipe
import core.api.consts.Items
import core.game.interaction.NodeUsageEvent
import core.game.node.item.Item

class PineapplePizza : PizzaRecipe() {

    override fun getExperience(): Double {
        return 52.0
    }

    override fun getLevel(): Int {
        return 65
    }

    override fun getProduct(): Item {
        return PINEAPPLE_PIZZA
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(PINEAPPLE_CHUNKS, PINEAPPLE_RING)
    }

    override fun getMixMessage(event: NodeUsageEvent): String {
        return "You add the " + event.baseItem.name.lowercase() + " to the pizza."
    }

    companion object {
        private val PINEAPPLE_PIZZA = Item(Items.PINEAPPLE_PIZZA_2301)
        private val PINEAPPLE_RING = Item(Items.PINEAPPLE_RING_2118)
        private val PINEAPPLE_CHUNKS = Item(Items.PINEAPPLE_CHUNKS_2116)
    }
}
